# Ex2 - Getting and Knowing your Data

# Step 1. Import the necessary libraries



In [4]:
import pandas as pd
import numpy as np

# Step 3. Assign it to a variable called chipo.

#tsv_file='chipotle.tsv'
#chipotle = pd.read_table(tsv_file,sep='\t')
#chipotle.to_csv('chipotle.csv',index=False)

In [6]:
chipotle = pd.read_csv("chipotle.csv")

In [7]:
chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98

# Step 4. See the first 10 entries

In [8]:
chipotle.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
6                                                NaN        1.69  
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75  
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...        9.25  
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...        9.25

# Step 5. What is the number of observations in the dataset?

In [9]:
#solution 1
chipotle.shape

(4622, 5)

In [10]:
#solution 2
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   int64  
 2   item_name           4622 non-null   object 
 3   choice_description  3376 non-null   object 
 4   item_price          4622 non-null   float64
dtypes: float64(1), int64(2), object(2)
memory usage: 180.7+ KB


# Step 6. What is the number of columns in the dataset?

In [11]:
#chipotle.columns
len(chipotle.columns)

5

# Step 7. Print the name of all the columns.

In [104]:
pd.DataFrame(chipotle.columns)

0
0            order_id
1            quantity
2           item_name
3  choice_description
4          item_price
5             revenue

# Step 8. How is the dataset indexed?

In [105]:
ind = chipotle.iloc[:,1]
ind
# chipotle.head(5)
# chipotle.info()

0       1
1       1
2       1
3       1
4       2
       ..
4617    1
4618    1
4619    1
4620    1
4621    1
Name: quantity, Length: 4622, dtype: int64

# Step 9. Which was the most-ordered item?

In [74]:
most_order_item = chipotle.groupby(by='item_name')['quantity'].sum().sort_values(ascending = False)
#most_order_item .index[0]
most_order_item

item_name
Chicken Bowl                             761
Chicken Burrito                          591
Chips and Guacamole                      506
Steak Burrito                            386
Canned Soft Drink                        351
Chips                                    230
Steak Bowl                               221
Bottled Water                            211
Chips and Fresh Tomato Salsa             130
Canned Soda                              126
Chicken Salad Bowl                       123
Chicken Soft Tacos                       120
Side of Chips                            110
Veggie Burrito                            97
Barbacoa Burrito                          91
Veggie Bowl                               87
Carnitas Bowl                             71
Barbacoa Bowl                             66
Carnitas Burrito                          60
Steak Soft Tacos                          56
6 Pack Soft Drink                         55
Chips and Tomatillo Red Chili Salsa       50


### Chicken Bowl( 761) was most-ordered item.                          

# Step 10. For the most-ordered item, how many items were ordered?

In [75]:
# most_order_item_total = chipotle.item_name.value_counts()  //the count of unique values in the chipotle
# max(most_order_item_total)                                 //max of unique values in chipotle

726

In [107]:
#chipotle[chipotle.item_name == 'Chicken Bowl']['quantity'].sum() #sum of total quantity of Item Chicken Bowl
chipotle[chipotle.item_name == 'Chicken Bowl'].quantity.sum()

761

In [69]:
max(chipotle.quantity)

15

# Step 11. What was the most ordered item in the choice_description column?

In [79]:
# most_order_Choice_description_total = chipotle.choice_description.value_counts()
# most_order_Choice_description_total.index[0]
# chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  revenue  
0                                                NaN        2.39     2.39  
1                                       [Clementine]        3.39     3.39  
2                                            [Apple]        3.39     3.39  
3                                                NaN        2.39     2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98    33.96

In [84]:
pd.DataFrame(chipotle.choice_description.value_counts()).head() #value_counts give unique values count.

choice_description
[Diet Coke]                                                        134
[Coke]                                                             123
[Sprite]                                                            77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese...                  42
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese...                  40

## The most ordered item in the choice_description column is Diet Coke(134).

# Step 12. How many items were orderd in total?

In [85]:
#most_order_Choice_description_total.sum()
chipotle.quantity.sum()  ## Total items order == sum(quantity)

4972

# Step 13. Turn the item price into a float
## Step 13.a. Check the item price type

In [18]:
print(chipotle.item_price.dtype)

float64


## Step 13.b. Create a lambda function and change the type of item price

In [19]:
chipotle['item_price'] = chipotle['item_price'].astype(str).str.replace('$','') 
chipotle.head()

<ipython-input-19-71ea7312cc56>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  chipotle['item_price'] = chipotle['item_price'].astype(str).str.replace('$','')


order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN       2.39  
1                                       [Clementine]       3.39  
2                                            [Apple]       3.39  
3                                                NaN       2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...      16.98

## Step 13.c. Check the item price type

In [20]:
#print(chipotle.item_price.dtype)
#chipotle['item_price'].astype(str).astype(float)
chipotle['item_price'] = pd.to_numeric(chipotle['item_price'])
print(chipotle.item_price.dtype)

float64


# Step 14. How much was the revenue for the period in the dataset?

In [29]:
chipotle.head(5)
revenue = (chipotle['quantity'] * chipotle['item_price']).sum()
revenue

39237.02

# Step 15. How many orders were made in the period?

In [94]:
#chipotle.order_id.value_counts.count()
len(chipotle.order_id.unique())  ## totale unique order

1834

In [108]:
chipotle.order_id.unique()

array([   1,    2,    3, ..., 1832, 1833, 1834], dtype=int64)

In [97]:
chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  revenue  
0                                                NaN        2.39     2.39  
1                                       [Clementine]        3.39     3.39  
2                                            [Apple]        3.39     3.39  
3                                                NaN        2.39     2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98    33.96

# Step 16. What is the average revenue amount per order?

In [100]:
# solution 1
#chipotle['revenue'] = chipotle['quantity'] * chipotle['item_price']
#avg_rev_per_order = chipotle.groupby(by='order_id')['revenue'].sum().mean()
#avg_rev_per_order
#chipotle = chipotle.drop(['avg_rev_per_order'],axis = 1)
#avg_rev_per_order.mean()

21.394231188658654

In [110]:
# solution 2
chipotle['revenue'] = chipotle['quantity'] * chipotle['item_price']
#chipotle.groupby(by='order_id')['revenue'].sum().mean()
chipotle.groupby(by='order_id').revenue.sum().mean()

21.394231188658654

# Step 17. How many different items are sold?

In [109]:
chipotle.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  revenue  
0                                                NaN        2.39     2.39  
1                                       [Clementine]        3.39     3.39  
2                                            [Apple]        3.39     3.39  
3                                                NaN        2.39     2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98    33.96

In [103]:
#chipotle.item_name.value_counts()
len(chipotle.item_name.unique())

50

In [26]:
chipotle.item_name.value_counts().count()

50

In [ ]:
chipotle

In [ ]:
chipotle.values

In [ ]:
chipotle.index

In [ ]:
chipotle.columns

In [ ]:
chipotle.info()

In [ ]:
chipotle.describe()

In [ ]:
chipotle.sort_values('quantity').head()